### PR TITLE
jwg12 IfcGetBasisSurface

### PR DESCRIPTION
**Pull request by SergejMuhic**
1. updated according to bSI instructions

EDIT: I would have at least preferred an "else if". Then you would have an easier time deciding whether to interpret it as a "switch case" or "else if".

### CODE REVIEW

File: IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometryResource/Functions/IfcGetBasisSurface/Expression.txt [hunk 1]
```diff
@@ -21,10 +21,22 @@
...
+
     IF N > 1 THEN
       REPEAT i := 2 TO N;
-        Surfs := Surfs * IfcGetBasisSurface(C\IfcCompositeCurve.Segments[1].ParentCurve);
+        IF ('IFCGEOMETRYRESOURCE.IFCCURVESEGMENT' IN TYPEOF(C\IfcCompositeCurve.Segments[i])) THEN
+          Surfs := Surfs * IfcGetBasisSurface(C\IfcCompositeCurve.Segments[i]\IfcCurveSegment.ParentCurve);
+        END_IF;
+        IF ('IFCGEOMETRYRESOURCE.IFCCOMPOSITECURVESEGMENT' IN TYPEOF(C\IfcCompositeCurve.Segments[i])) THEN
+          Surfs := Surfs * IfcGetBasisSurface(C\IfcCompositeCurve.Segments[i]\IfcCompositeCurveSegment.ParentCurve);
+        END_IF;
```
Comment (pjanck) [inline IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometryResource/Functions/IfcGetBasisSurface/Expression.txt:39]: In deleted line, there is `Segments[1]`, in the additions, it is `Segments[i]`. Intentional fix?

Comment (SergejMuhic) [reply IFC4x3/Sections/Resource definition data schemas/Schemas/IfcGeometryResource/Functions/IfcGetBasisSurface/Expression.txt:39]: Yes, that was a mistake to be corrected if I remember correctly. Old one.

**Review by pjanck — APPROVED** [review]
Not the nicest of syntax (agreeing with the edit note), but so be it.

**Comment by SergejMuhic** [comment]
@TLiebich any opinions? I think we should at least make it an "If else if". But if we do it here, we have to do it on bSI also. I am in no position to do that.

**Comment by TLiebich** [comment]
I would leave it as is (of cause including the differentiation of curve segement and composite curve segment. We inherited it from STEP, I just checked the late(st) publication and it is not altered there.

**Pull request merged**